distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/model/AssignmentState.java

```diff
@@ -0,0 +1,185 @@
+     */
+    public WorkerId getLessBusyWorkerFromAvailableWorkers(Set<WorkerId> availableWorkers) {
+        WorkerId lessBusyWorker = null;
+        int minWorkPool = Integer.MAX_VALUE;
+
+        for (Map.Entry<WorkerId, HashSet<WorkItem>> worker : entrySet()) {
+            if (!availableWorkers.contains(worker.getKey())) {
+                continue;
+            }
+            HashSet<WorkItem> workPool = worker.getValue();
+
+            if (workPool.size() < minWorkPool) {
+                minWorkPool = workPool.size();
+                lessBusyWorker = worker.getKey();
+            }
+        }
+        return lessBusyWorker;
+    }
+
+    /**
+     * @return worker which has most work pool size (doesn't depends on job)
+     */
+    public WorkerId getMostBusyWorker() {
+        WorkerId mostBusyWorker = null;
+        int minWorkPool = Integer.MIN_VALUE;
+
+        for (Map.Entry<WorkerId, HashSet<WorkItem>> worker : entrySet()) {
+            HashSet<WorkItem> workPool = worker.getValue();
+
+            if (workPool.size() > minWorkPool) {
+                minWorkPool = workPool.size();
+                mostBusyWorker = worker.getKey();
+            }
+        }
+        return mostBusyWorker;
+    }
+
+    public boolean containsWorkItem(WorkItem workItem) {
+        for (Map.Entry<WorkerId, HashSet<WorkItem>> worker : entrySet()) {
+            for (WorkItem item : worker.getValue()) {
+                if (workItem.equals(item)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public boolean containsAnyWorkItemOfJob(WorkerId workerId, JobId jobId) {
+        for (WorkItem item : get(workerId)) {
+            if (jobId.getId().equals(item.getJobId())) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public boolean containsWorkItemOnWorker(WorkerId workerId, WorkItem workItem) {
+        for (WorkItem item : get(workerId)) {
+            if (workItem.equals(item)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return worker on which work item placed
+     */
+    public WorkerId getWorkerOfWorkItem(WorkItem workItem) {
+        for (Map.Entry<WorkerId, HashSet<WorkItem>> worker : entrySet()) {
+            for (WorkItem work : worker.getValue()) {
+                if (workItem.equals(work)) {
+                    return worker.getKey();
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @return true, if work pool sizes of various workers differ more than 1
+     * For example:
+     * worker-1: 4
+     * worker-2: 5
+     * worker-3: 5
+     * is balanced, returns true
+     * worker-1: 0
+     * worker-2: 0
+     * returns also true
+     * worker-1: 4
+     * worker-2: 5
+     * worker-3: 3
+     * returns false
+     */
+    public boolean isBalanced() {
```

Comment:
It can be useful in custom assignment strategies.
For example:
```
while (!state.isBalanced()) { 
    WorkerId lessBusy = state.getLessBusyWorker(); 
    WorkerId mostBusy = state.getMostBusyWorker(); 
    WorkItem workItem = ...
    moveWorkItem(workItem, mostBusy, lessBusy);
} 
```